### PR TITLE
032 improve postman parser

### PR DESCRIPTION
**Pull request by JonathanMontane**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -51,4 +51,4 @@ crashlytics-build.properties
 build/
 node_modules/
 lib/
-
+.DS_Store
```
Comment (JonathanMontane) [inline .gitignore]: Critical change

File: src/importers/postman/Parser.js [hunk 2]
```diff
@@ -413,19 +427,21 @@ export default class PostmanParser {
...
+                    queries.push(new KeyValue({
+                        key: this._referenceEnvironmentVariable(
+                            decodeURIComponent(m[1])
+                        ),
+                        value: typeof m[2] === 'string' ?
+                            this._referenceEnvironmentVariable(
+                                decodeURIComponent(m[2])
+                            ) : null
+                    }))
+                }
```
Comment (JonathanMontane) [inline src/importers/postman/Parser.js]:
```
if (m) {
   ...
}
```

is necessary to filter edge cases such as:
`https://url.com/weird?`

File: src/importers/postman/Parser.js [hunk 3]
```diff
@@ -534,6 +551,24 @@ export default class PostmanParser {
...
+                        'children', req.get('name') ||
+                        req.get('url')
+                    ],
+                    req
+                )
+            }
+        }
+
+        return _group
+    }
```
Comment (JonathanMontane) [inline src/importers/postman/Parser.js]:
```
if (req) {
   ...
}
```

is necessary for the edge case where the folder data from the collection is corrupted, and contains non existing request ids

File: src/importers/postman/Parser.js (hunk 2 above)
Comment (mittsh) [inline src/importers/postman/Parser.js]: what will happen here with such a URL?

File: src/importers/postman/Parser.js [hunk 4]
```diff
@@ -499,7 +516,7 @@ export default class PostmanParser {
         }
         else if (req.dataMode === 'urlencoded' || req.dataMode === 'params') {
             if (req.dataMode === 'urlencoded') {
-                bodyType = 'urlEncode'
+                bodyType = 'urlEncoded'
```
Comment (mittsh) [inline src/importers/postman/Parser.js]: ouch!

File: src/importers/postman/Parser.js (hunk 2 above)
Comment (mittsh) [inline src/importers/postman/Parser.js]: alright… yeah I got it, you fixed it with the `if` statement… good catch

File: src/importers/postman/Parser.js [hunk 5]
```diff
@@ -386,9 +386,23 @@ export default class PostmanParser {
             hawkAuth: ::this._extractHawkAuth
         }
 
+        let _helper = helper
+        if (typeof _helper === 'string') {
+            try {
+                _helper = JSON.parse(_helper)
```
Comment (mittsh) [inline src/importers/postman/Parser.js]: I guess this is the heart of the fix! :)

**Comment by mittsh** [comment]
👍 great!

**Pull request merged**